### PR TITLE
Update authentication to use OAuth

### PR DESCRIPTION
**Pull request by craibo**
Panasonic updated the authentication process to OAuth
- Initial release to get the HA integration working again for users
- Refactor the Authentication logic based on the work here https://github.com/little-quokka/python-panasonic-comfort-cloud
- Additional work is required to clean up and improve on these changes
- Importantly the logic need to be update to use the HA OAuth flow which will reduce the authentication process and use the refresh token instead

### CODE REVIEW

**Comment by danielcherubini** [comment]
I have tested this and found no issues since the `2024.6.2` release.

Upgrading existing works fine, and also installing it, and setting it up as a new integration works fine.

**Comment by sockless-coding** [comment]
This code breaks the auto app version, this is the most common way Panasonic blocks the integration.

**Comment by danielcherubini** [comment]
@sockless-coding you're best to check out https://github.com/lostfields/python-panasonic-comfort-cloud/pull/94 where the next rewrite of the API requests are being worked on.

The idea is once that's merged, then @craibo will use that from here on out.

**Comment by danielcherubini** [comment]
https://github.com/lostfields/python-panasonic-comfort-cloud/pull/94/files#diff-05be53fe625379891b3f62b48483947e8939772bbe4d8acfa1a720258e45152dR243 

That's the specific line you're talking about

**Comment by danielcherubini** [comment]
@sockless-coding I've made the changes over in https://github.com/lostfields/python-panasonic-comfort-cloud/pull/94

@craibo the changes are ready for you.

**Comment by sockless-coding** [comment]
I managed to break something with the authentication when switching to async... I'll look at it in an hour or so unless someone fixes before that

**Comment by danielcherubini** [comment]
> I managed to break something with the authentication when switching to async... I'll look at it in an hour or so unless someone fixes before that

It's probably the appversion.

File: custom_components/panasonic_cc/config_flow.py [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
 import asyncio
-from typing import Any, Dict, Optional
 import logging
+from typing import Any, Dict, Optional
 
-from aiohttp import ClientError
-from async_timeout import timeout
 import voluptuous as vol
-
+from aiohttp import ClientError
```
Comment (danielcherubini) [inline custom_components/panasonic_cc/config_flow.py:7]: @sockless-coding please add aiohttp to requirements.txt

**Comment by danielcherubini** [comment]
To explain more, since we cannot reverse the iOS app calls, to get the iOS app client credentials, we need to use the android app version. Since that's the version that's tied to the OAuth2 checks.

iOS = 1.20.1
Android = 1.20.0

There is no API to get the info from the GooglePlay Store. The only way is with api credentials, and that costs money if you go over a number of API calls... I don't think it's smart to tie anyone's creditcard to this. So I'm working on a way to publish the Android API version number to a public gist when it changes. 

if you update your check for the version number to something similar to this, then it will work with what https://github.com/sockless-coding/panasonic_cc is going to do.

```python
def _update_app_version(self):

        if self._raw:
            print("--- auto detecting latest app version")
        try:
            response = requests.get(PanasonicSession.VERSION_GIST)
            data = json.loads(response.text)
            version = data['files']['comfort-cloud-version']['content']
           if version is not None:
                _LOGGER.debug(f"Found app version: {version}")
                self._appVersion = version
                self._settings.version = version
                return
        except Exception as e:
            _LOGGER.warning(f"Getting app version: {e}")
            pass
        _LOGGER.debug(f"Failed to retrive app version using version {self._appVersion}")
```

**Comment by danielcherubini** [comment]
https://github.com/danielcherubini/comfort-cloud-version-updater
Here's the version updater if anyone is curious, I'm going to deploy this now and put it on a 6h cron job

**Comment by sockless-coding** [comment]
Yeah, you are right it's the version. But I don't think we need to maintain our own version scraper. AppBrain seems to do that already.
https://www.appbrain.com/app/panasonic-comfort-cloud/com.panasonic.ACCsmart so I just change it to point to that

**Comment by danielcherubini** [comment]
@sockless-coding I thought about that, but didn't know how often appbrain would update... I imagine it's almost instantly.

**Comment by danielcherubini** [comment]
Although.... they have a meta element that seems like it would be pretty safe to scrape

`<meta itemprop="softwareVersion" content="1.20.0">`

**Comment by sockless-coding** [comment]
Yeah scraping would probaly work fine

**Comment by danielcherubini** [comment]
Here you go @sockless-coding 

```python
 def _update_app_version(self):
        if self._raw:
            print("--- auto detecting latest app version")
        try:
            response = requests.get(PanasonicSession.APPBRAIN_URL)
            responseContent = response.content
            soup = BeautifulSoup(responseContent, "html.parser")
            meta_tag = soup.find("meta", itemprop="softwareVersion")
            if meta_tag:
                version = meta_tag['content']
                _LOGGER.debug(f"Found app version: {version}")
                self._app_version = version
                # self._settings.version = version
                if self._raw:
                    print("--- found version: {}".format(self._app_version))
                return

        except Exception as e:
            print(f"Error Get version: {e}")
            pass
        if self._raw:
            print(
                "--- failed to detect app version using version: {}".format(self._app_version))

```

**Comment by danielcherubini** [comment]
I'll convert requests for aiohttp in the upstream library too.

**Comment by codyc1515** [comment]
Works for me. Thank you very much!

**Comment by marnixt** [comment]
thanks very much for the update to fix the authentication issues! Much appreciated!

**Pull request merged**